spec/requests/api/v3.1/app_meta_data_spec.rb

```diff
@@ -0,0 +1,37 @@
+RSpec.describe("v3.1 - AppMetaData") do
+  include ::Spec::Support::TenantIdentity
+
+  let(:messaging_client) { instance_double("ManageIQ::Messaging::Client") }
+  before do
+    allow(messaging_client).to receive(:publish_topic)
+    allow(Sources::Api::Messaging).to receive(:client).and_return(messaging_client)
+  end
+
+  let(:headers)          { {"CONTENT_TYPE" => "application/json", "x-rh-identity" => identity} }
+  let(:attributes)       { {"name" => "a_field", "payload" => {"value" => 1234}} }
```

Comment:
This question may be off base - but if we have an STI class AppMetaData - wouldn't the payload include a "type" key? - Maybe it's supressed in openapi spec and this question is mute.  

I'm coming from the perspective of a DB lookup:  `AppMetaData.last` would return the `:type` attribute `==` `app_meta_data`. So I'm wondering if the returned json payload would include a "type" key?

Reply:
Yep you're right, it won't unless we expose it in the openapi.json. We can expose it if we want to, but this is the only "user facing" metadata so a type doesn't quite make sense unless we expand it. 

Current output here:
```
http :3002/api/sources/v3.1/app_meta_data $rhid --body
{
    "data": [
        {
            "application_type_id": "2",
            "created_at": "2021-01-26T21:08:09Z",
            "id": "81",
            "name": "gcp_service_account",
            "payload": "test-billing-service-account@cloud-billing-292519.iam.gserviceaccount.com",
            "updated_at": "2021-01-26T21:08:09Z"
        }
    ],
    "links": {
        "first": "/api/sources/v3.1/app_meta_data?offset=0",
        "last": "/api/sources/v3.1/app_meta_data?offset=0"
    },
    "meta": {
        "count": 1,
        "limit": 100,
        "offset": 0
    }
}
```

Reply:
No, I'm fine with it not being exposed - unless the SuperKeyMetaData would be exposed: In that case the type would be helpful and/or would most likely be a required field for a POST.

For now this makes sense to me.